spec/config-spec.ts

```diff
@@ -1,9 +1,168 @@
+/** @prettier */
+
 import { config } from '../src/internal/config';
 import { expect } from 'chai';
+import { Observable } from 'rxjs';
 
 describe('config', () => {
   it('should have a Promise property that defaults to nothing', () => {
     expect(config).to.have.property('Promise');
     expect(config.Promise).to.be.undefined;
   });
+
+  describe('onUnhandledError', () => {
+    afterEach(() => {
+      config.onUnhandledError = null;
+    });
+
+    it('should default to null', () => {
+      expect(config.onUnhandledError).to.be.null;
+    });
+
+    it('should call asynchronously if an error is emitted and not handled by the consumer observer', (done) => {
+      let called = false;
+      const results: any[] = [];
+
+      config.onUnhandledError = err => {
+        called = true;
+        expect(err).to.equal('bad');
+        done()
+      };
+
+      const source = new Observable<number>(subscriber => {
+        subscriber.next(1);
+        subscriber.error('bad');
+      });
+
+      source.subscribe({
+        next: value => results.push(value),
+      });
+      expect(called).to.be.false;
+      expect(results).to.deep.equal([1]);
+    });
+
+    it('should call asynchronously if an error is emitted and not handled by the consumer next callback', (done) => {
```

Comment:
nitpick: the description is weird; the consumer's `next` callback cannot handle the error